pkg/ipmasq/cell/cell.go

```diff
@@ -0,0 +1,76 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package cell
+
+import (
+	"fmt"
+	"log/slog"
+
+	"github.com/cilium/cilium/pkg/ipmasq"
+	ipmasqmaps "github.com/cilium/cilium/pkg/maps/ipmasq"
+	"github.com/cilium/cilium/pkg/option"
+
+	"github.com/cilium/hive/cell"
+)
+
+var Cell = cell.Module(
+	"ip-masq-agent",
+	"BPF ip-masq-agent implementation",
+
+	cell.Provide(newIPMasqAgentCell),
+	cell.Invoke(registerIPMasqAgentLifecycle),
+	cell.Config(defaultConfig),
+)
+
+type ipMasqAgentParams struct {
+	cell.In
+
+	Logger    *slog.Logger
+	Lifecycle cell.Lifecycle
+	Config    Config
+	IPMasqMap *ipmasqmaps.IPMasqBPFMap
+}
+
+type ipMasqAgentOut struct {
+	cell.Out
+
+	IPMasqAgent *ipmasq.IPMasqAgent
+}
+
+func newIPMasqAgentCell(params ipMasqAgentParams) (ipMasqAgentOut, error) {
+	if !option.Config.EnableIPMasqAgent {
+		return ipMasqAgentOut{}, nil
+	}
+
+	agent := ipmasq.NewIPMasqAgent(params.Logger, params.Config.IPMasqAgentConfigPath, params.IPMasqMap)
+
+	return ipMasqAgentOut{
+		IPMasqAgent: agent,
+	}, nil
+}
```

Comment:
Nit: You can directly return *ipmasq.IPMasqAgent type from this constructor instead of using cell.Out embedded struct with single member.